core/chains/evm/client/errors.go

```diff
@@ -353,9 +361,15 @@ func (s *SendError) IsServiceUnavailable(configErrors *ClientErrors) bool {
 	return s.is(ServiceUnavailable, configErrors)
 }
 
-// IsOutOfCounters is a zk chain specific error returned if the transaction is too complex to prove on zk circuits
-func (s *SendError) IsOutOfCounters(configErrors *ClientErrors) bool {
-	return s.is(OutOfCounters, configErrors)
+// IsTerminallyStuck indicates if a transaction was stuck without any chance of inclusion
+func (s *SendError) IsTerminallyStuckConfigError(configErrors *ClientErrors) bool {
+	return s.is(TerminallyStuck, configErrors)
+}
+
+// Naming discrepancy is due to the generic transaction states introduced by ChainWriter
+// Errors classified as terminally stuck are considered fatal since the transaction payload should NOT be retried by external callers
+func (s *SendError) IsFatal() bool {
```

Comment:
Not sure if this is a good idea. There is already a `Fatal` method at the top. Adding `IsFatal` is a bit confusing to me. Shouldn't we aggregate those? Doesn't the CW want to know about all the Fatal cases? Why the distinction here?

Reply:
I think there's a distinction between what's fatal/retryable at the TXM level and what's fatal/retryable(failed) at the product level. `SendError` classifies these errors for the TXM level so something that's classified as retryable here doesn't necessarily require any action from the product side. The only time they need to take action is once the transaction is in a completed state like `fatal_error`. The CW errors require us to split the reason for `fatal_error` into two. This is sort of where the confusion is introduced. What's `fatal` from the product's perspective is ZK overflow (terminally stuck) errors. The rest are considered `failed` which are classified as fatal in SendError. 
I think I'm probably making things confusing here by trying to leverage `SendError` errors for CW error classification. I considered creating a wrapper but felt it was adding more bloat when we could already idenitfy ZK overflow through it.

Reply:
I think I could rename this and also update the `ErrorClassifier` [type](https://github.com/smartcontractkit/chainlink/pull/13040/files#diff-93cc93f19e9c89ed8063779fdac239db0cd4dadfcbd9484a6b9e2b8a2ef2743dR345-R348) to make it less confusing. Would you be ok with `IsFatalForProducts` or I could make it CW specific with `IsChainWriterFatal`